jobs/ScheduleBackupJob.js

```diff
@@ -80,11 +80,42 @@ class ScheduleBackupJob extends BaseJob {
   }
 
   static deleteOldBackup(job, instanceDeleted) {
+    let transactionLogsDeleteBefore;
+
+    function filterOldBackups(oldBackups) {
+      let filteredOldBackups = [];
+      /* oldBackups : This aray should contain all older backups
+      // including last retenion day's backup. E.g. if retention period
+      // is 15 days it would include all backups in (15th, 16th, 17th ...) */
+      if (typeof oldBackups !== 'undefined' && oldBackups.length > 0) {
+        // Older backups are sorted as latest at first
+        let sortedBackups = _.sortBy(oldBackups, ['started_at']).reverse();
+        const latestSuccessIndex = _.findIndex(sortedBackups,
+          backup => backup.state === CONST.OPERATION.SUCCEEDED);
+        if (latestSuccessIndex === -1) {
+          //No successful backup beyond retention period.
+          filteredOldBackups = sortedBackups;
+          transactionLogsDeleteBefore = new Date(Date.now() - (config.backup.retention_period_in_days + 1) * 24 * 60 * 60 * 1000).toISOString();
```

Comment:
Also handle the case when instance deleted. In that case after retention period post instance deletion, all backups and transaction logs should be cleaned up.

Reply:
Won't that be automatically taken care by above logic? Since, after the retention period, the backups will be deleted and upon not finding any backup, the above logic kicks in and deletes transaction logs older than 15 days.